Tests/KlaviyoSwiftTests/StateManagementEdgeCaseTests.swift

```diff
@@ -369,15 +369,30 @@ class StateManagementEdgeCaseTests: XCTestCase {
     // MARK: - set enqueue event uninitialized
 
     @MainActor
-    func testEnqueueEventUninitialized() async throws {
+    func testOpenedPushEventUninitializedAddsToPendingRequests() async throws {
+        let store = TestStore(initialState: .init(queue: []), reducer: KlaviyoReducer())
+        let event = Event(name: .OpenedPush)
+        _ = await store.send(.enqueueEvent(event)) {
+            $0.pendingRequests = [.event(event)]
+        }
+    }
+
+    @MainActor
+    func testEnqueueNonOpenedPushEventUninitializedDoesNotAddToPendingRequest() async throws {
```

Comment:
This test goes through all other event metrics other than opened push and makes sure that they don't get through the SDK when it's not initialized.

Reply:
Nice.